src/averageAges.js

```diff
@@ -55,7 +69,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const arrayOfChildren = onlyWithSon
+    ? people
+      .filter(person => people.some(woman => person.sex === 'm'
+                                  && person.mother === woman.name))
+    : people
+      .filter(person => people.some(woman => person.mother === woman.name));
+
+  const arrayOfMothers = onlyWithSon
```

Comment:
This variable is redundant. In "ageDifferences" you can use the name of the initial array instead of this one.